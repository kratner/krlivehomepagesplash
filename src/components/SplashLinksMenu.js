import React, { Component } from "react";
import { connect } from "react-redux";
import { getSplashLinks } from "../actions/SplashLinksActions";
import styled from "styled-components";
import { CircularProgress } from "@material-ui/core";

const SplashLinksMenuContainer = styled.div``;

class SplashLinksMenu extends Component {
  componentDidMount() {
    this.props.dispatch(getSplashLinks());
  }
  render() {
    let listItems = "";
    if (this.props.hasListItems) {
      listItems = this.props.listItems.map((node, index) => {
        return (
          <li key={index}>
            <a
              href={node.href}
              target="_blank"
              rel="noopener noreferrer"
              title={node.title}
            >
              {node.linkText}
            </a>
          </li>
        );
      });
    }
    return (
      <SplashLinksMenuContainer id={this.props.id}>
        {this.props.loading ? (
          <div>
            <CircularProgress />
          </div>
        ) : (
          ""
        )}
        {this.props.hasListItems ? <ul>{listItems}</ul> : ""}
      </SplashLinksMenuContainer>
    );
  }
}

const mapStateToProps = state => {
  return {
    listItems: state.splashLinks.listItems,
    hasListItems: state.splashLinks.hasListItems,
    loading: state.splashLinks.loading
  };
};

export default connect(mapStateToProps)(SplashLinksMenu);
