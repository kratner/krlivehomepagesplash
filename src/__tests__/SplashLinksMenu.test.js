import React from "react";
import { shallow } from "enzyme";
import SplashLinksMenu from "../components/SplashLinksMenu";
import thunk from "redux-thunk";
import { createStore, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import reducer from "../reducers/index";

const store = createStore(reducer, applyMiddleware(thunk));

describe("SplashLinksMenu", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <SplashLinksMenu />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
