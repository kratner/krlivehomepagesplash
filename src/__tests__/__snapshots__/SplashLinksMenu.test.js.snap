// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashLinksMenu renders correctly 1`] = `ShallowWrapper {}`;
